logic.py

```diff
@@ -998,4 +1000,76 @@ def export_workflow_report(article_list, averages):
 
         all_rows.append(row)
 
-    return export_csv(all_rows)
\ No newline at end of file
+    return export_csv(all_rows)
+
+
+def get_metrics_start_end(request):
+    """
+    Should be used in combination with DateRangeForm. Fetches start_date and
+    end_date from request.GET and transforms them into datetimes better
+    suited for filtering accesses.
+    """
+    try:
+        try:
+            # Get start and end date from request.GET
+            start_date = datetime.strptime(
+                request.GET.get('start_date'),
+                '%Y-%m-%d',
+            )
+            end_date = datetime.strptime(
+                request.GET.get('end_date'),
+                '%Y-%m-%d',
+            ).replace(
+                hour=23,
+                minute=59,
+                second=59,
+            )
+        except ValueError:
+            start_date, end_date = None, None
+            messages.add_message(
+                request,
+                messages.WARNING,
+                'Date not in recognised format Y-m-d',
+            )
+    except TypeError:
+        # No defaults supplied at all, set to None and generate below.
+        start_date, end_date = None, None
+
+    if not start_date and not end_date:
```

Comment:
Shall we populate the defaults for start_date and end_date to avoid having to catch the TypError nested with ValueError. Would make this easier to read and is what we usually do for most functions

Reply:
Change added.